pkg/probe/http/http.go

```diff
@@ -70,6 +70,7 @@ type httpProber struct {
 
 // Probe returns a ProbeRunner capable of running an HTTP check.
 func (pr httpProber) Probe(url *url.URL, headers http.Header, timeout time.Duration) (probe.Result, string, error) {
+	pr.transport.DisableCompression = true // removes Accept-Encoding header
```

Comment:
interesting... So you disable the header using `DisableCompression`, but can add header by simply adding it to the headers collection. After a brief internet search I wonder if Accept-Encoding needs to be set to `identity` as well (https://groups.google.com/g/golang-nuts/c/ms8QNCzew8Q)? Cannot find any call to dumping http request anywhere, maybe you saw it somewhere?

Reply:
`If no Accept-Encoding field is in the request, any content-coding is considered acceptable by the user agent.`

We should not need to set "Accept-Encoding: identity", in fact my understanding is that having no Accept-Encoding is different from "Accept-Encoding: identify", as with the former should be equivalent to "Accept-Encoding: *" 
~> "give-me no-compression" vs "give-me anything" (?)

( https://tools.ietf.org/html/rfc7231#section-5.3.4 )

Also to me, it seems we may have 2 different questions:
* do we just want to reduce the payload ?
* do we want to disable the compression by default ?

Reply:
got it. Thank you. I misunderstood the article than,

With your explanation I think removing the headers benefits both - less payload on request and response can still be gzip for apps worried about network and non-gzip for ones that are CPU bound.

Reply:
yeah, i think so. Seems server should fallback to gzip if no "identify" is available, so basically means that it should not break in case the server only knows to respond with gzip encoding 🤔 

> Note: If the request does not include an Accept-Encoding field,
>       and if the "identity" content-coding is unavailable, then
>       content-codings commonly understood by HTTP/1.0 clients (i.e.,
>       "gzip" and "compress") are preferred; some older clients
>       improperly display messages sent with other content-codings.  The
>       server might also make this decision based on information about
>       the particular user-agent or client

https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html

Reply:
*aside* If you want to go wild, send `TE: gzip` and allow HTTP/1.1 gzip _transfer encoding_.
(Not many origins support that). More details in https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/TE